src/payment/strategies/bluesnapv2/bluesnapv2-payment-strategy.ts

```diff
@@ -0,0 +1,68 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import * as paymentStatusTypes from '../../payment-status-types';
+import PaymentStrategy from '../payment-strategy';
+
+import BlueSnapV2PaymentPageLoader from './bluesnapv2-payment-page-loader';
+
+export default class BlueSnapV2PaymentStrategy implements PaymentStrategy {
+
+    private readonly TARGET = 'bluesnapv2_hosted_payment_page';
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _blueSnapV2PaymentPageLoader: BlueSnapV2PaymentPageLoader
+    ) {}
+
+    execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment } = orderRequest;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        return this._store.dispatch(
+            this._orderActionCreator.submitOrder(orderRequest, options)
+        )
+        .then(() =>
+            Promise.race([
```

Comment:
don't you need both things to finish? Wouldn't Promise.race return as soon as any of those have resolved?

Reply:
Actually, initializeOffsitePayment never finish
https://github.com/bigcommerce/checkout-sdk-js/blob/58bef4da9d7ec189761ff5b9a76e0e53a234c1a3/src/payment/payment-action-creator.ts#L57
because of the initializeOffsitePayment  from PaymentRequestSender don't solve the promise, due to the offsite approach
https://github.com/bigcommerce/checkout-sdk-js/blob/58bef4da9d7ec189761ff5b9a76e0e53a234c1a3/src/payment/payment-request-sender.ts#L30
so I only reject it when the modal gets closed.